src/common/libflux/fripp.c

```diff
@@ -341,84 +368,107 @@ static void timer_cb (flux_reactor_t *r,
 
 void fripp_set_agg_period (struct fripp_ctx *ctx, double period)
 {
+    if (!fripp_enabled (ctx))
+        return;
+
     if (period <= 0) {
         flux_watcher_stop (ctx->w);
         ctx->period = 0;
         return;
     }
 
     ctx->period = period;
+
     double after = flux_watcher_next_wakeup (ctx->w) - (double) time (NULL);
 
     flux_timer_watcher_reset (ctx->w, after, ctx->period);
 }
 
-void fripp_ctx_destroy (void *arg)
+void fripp_ctx_destroy (struct fripp_ctx *ctx)
 {
-    struct fripp_ctx *ctx = arg;
-
-    if (ctx->w) {
-        flux_watcher_stop (ctx->w);
+    if (ctx) {
+        int saved_errno = errno;
         flux_watcher_destroy (ctx->w);
-    }
-    if (ctx->metrics) {
-        zhashx_purge (ctx->metrics);
-        zhashx_destroy (&ctx->metrics);
-    }
-    if (ctx->done) {
+        if (ctx->metrics) {
+            zhashx_purge (ctx->metrics);
```

Comment:
It appears `zhashx_destroy()` internally calls purge (minus the attempt to shrink the hash table), so perhaps this line could be removed? (Sorry I missed that in the review of the initial fripp pr)